### PR TITLE
[exporter/elasticsearch] Translate `k8s.*.name` resource attributes in ECS mode

### PR DESCRIPTION
**Pull request by carsonip**
#### Description

In ECS mode:
Translate `k8s.job.name`, `k8s.cronjob.name`, `k8s.statefulset.name`, `k8s.replicaset.name`, `k8s.daemonset.name`, `k8s.container.name` to `kubernetes.*.name`. Translate `k8s.cluster.name` to `orchestrator.cluster.name`.

### CODE REVIEW

**Review by gizas — APPROVED** [review]

File: exporter/elasticsearchexporter/model.go [hunk 1]
```diff
@@ -57,6 +57,11 @@ var resourceAttrsConversionMap = map[string]string{
 	semconv.AttributeK8SNodeName:            "kubernetes.node.name",
 	semconv.AttributeK8SPodName:             "kubernetes.pod.name",
 	semconv.AttributeK8SPodUID:              "kubernetes.pod.uid",
+	semconv.AttributeK8SJobName:             "kubernetes.job.name",
+	semconv.AttributeK8SCronJobName:         "kubernetes.cronjob.name",
+	semconv.AttributeK8SStatefulSetName:     "kubernetes.statefulset.name",
+	semconv.AttributeK8SReplicaSetName:      "kubernetes.replicaset.name",
+	semconv.AttributeK8SDaemonSetName:       "kubernetes.daemonset.name",
```
Comment (ChrsMark) [inline exporter/elasticsearchexporter/model.go:64]: Shall we also map the `k8s.container.name` too? 

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/528678e607b60b00cbb5432b7ba8d2fcdc8cb810/processor/k8sattributesprocessor/config.go#L146

Comment (carsonip) [reply exporter/elasticsearchexporter/model.go:64]: added cluster name and container name for consistency.

**Review by ChrsMark — APPROVED** [review]

File: exporter/elasticsearchexporter/model.go [hunk 2]
```diff
@@ -57,6 +57,13 @@ var resourceAttrsConversionMap = map[string]string{
...
+	semconv.AttributeK8SClusterName:         "kubernetes.cluster.name",
```
Comment (tetianakravchenko) [inline exporter/elasticsearchexporter/model.go:65]: @carsonip there is no such field as `kubernetes.cluster.name`, it should be removed

Comment (carsonip) [reply exporter/elasticsearchexporter/model.go:65]: TODO to self: confirm if it is needed, as it is not in https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/otel-data/src/main/resources/component-templates/semconv-resource-to-ecs%40mappings.yaml

Comment (gregkalapos) [reply exporter/elasticsearchexporter/model.go:65]: Should it maybe point to `orchestrator.cluster.name`? 

From https://github.com/elastic/integrations/blob/main/packages/kubernetes/data_stream/container_logs/fields/ecs.yml#L22 . As @tetianakravchenko pointed out over slack - those fields are covered by ECS under `orchestrator.*` and not under `kubernetes.*`. 

That's the reason why it's not in  https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/otel-data/src/main/resources/component-templates/semconv-resource-to-ecs%40mappings.yaml.

Comment (ChrsMark) [inline exporter/elasticsearchexporter/model.go:65]:
```suggestion
	semconv.AttributeK8SClusterName:         "orchestrator.cluster.name",
```

ref: https://www.elastic.co/guide/en/ecs/current/ecs-orchestrator.html#field-orchestrator-cluster-name

File: exporter/elasticsearchexporter/model_test.go [hunk 3]
```diff
@@ -444,7 +451,14 @@ func TestEncodeLogECSMode(t *testing.T) {
...
 		"kubernetes.pod.name":        "opentelemetry-pod-autoconf",
 		"kubernetes.pod.uid":         "275ecb36-5aa8-4c2a-9c47-d8bb681b9aff",
-		"kubernetes.deployment.name": "coredns"
+		"kubernetes.deployment.name": "coredns",
+		"kubernetes.job.name":         "job.name",
+		"kubernetes.cronjob.name":     "cronjob.name",
+		"kubernetes.statefulset.name": "statefulset.name",
+		"kubernetes.replicaset.name":  "replicaset.name",
+		"kubernetes.daemonset.name":   "daemonset.name",
+		"kubernetes.cluster.name":     "cluster.name",
```
Comment (ChrsMark) [inline exporter/elasticsearchexporter/model_test.go:460]:
```suggestion
		"orchestrator.cluster.name":     "cluster.name",
```

ref: https://www.elastic.co/guide/en/ecs/current/ecs-orchestrator.html#field-orchestrator-cluster-name

File: exporter/elasticsearchexporter/model.go (hunk 2 above)
Comment (carsonip) [reply exporter/elasticsearchexporter/model.go]: Updated

**Review by tetianakravchenko — APPROVED** [review]

**Review by lahsivjar — APPROVED** [review]

**Review by ChrsMark — APPROVED** [review]

**Pull request merged**